pkg/networkservice/connectioncontext/dnscontext/vl3dns/server.go

```diff
@@ -81,10 +88,19 @@ func NewServer(chanCtx context.Context, getDNSServerIP func() net.IP, opts ...Op
 
 	result.listenAndServeDNS(chanCtx, result.dnsServer, fmt.Sprintf(":%v", result.dnsPort))
 
+	select {
+	case ip := <-getDNSServerIP:
+		result.dnsServerIP = ip
+	default:
+	}
```

Comment:
```suggestion
     go func() {
     
          for {
               select {
                    case <-ctx.Done():
                       return
                    case addr, ok := <- r. serverAddressCh:
                      if !ok {
                         return
                      }
                       updateServerAddress(ctx, addr)
                 }
          }
     }()
```

Reply:
This won't work because we doesn't have metadata and monitorConnection when we create server